src/agent.rs

```diff
@@ -322,8 +325,8 @@ impl AgentBuilder {
 
     /// Attempt to detect proxy settings from the environment, i.e. HTTP_PROXY
     ///
-    /// The default is `false`, i.e. not detecting proxy from env since this is
-    /// a potential security risk.
+    /// The default is `false` without the `proxy-from-env` feature flag, i.e.
+    /// not detecting proxy from env, to maintain backward compatibility.
```

Comment:
Having automatic proxy detection _is_ a security risk. I want you to rephrase this doc to keep the warning.